### DISCUSSION TITLE
Data export pattern

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1806] [Q&A] [answered]
Hiya, 

I've been reading over the Data Export pattern and I'm not super confident that the term 'Export' is the right option for all immediate downloads. 

In the below example, 'Export' denotes that I want to transform the data in the table into CSV format. Whereas, 'Download' is to denote that I want to download a document (in this case, the invoice) that is associated with the month. 'Export Invoice', I think would confuse the user and to me, sounds wrong. The invoice is currently not on the page they're viewing - so what would they be exporting?

![Usage   Invoices 1](https://user-images.githubusercontent.com/72031986/130821469-51f4f347-6e9d-420d-a345-15166eaa1303.png)

It would also be inconsistent with the console where they can 'Download' the invoice. 
![image](https://user-images.githubusercontent.com/72031986/130821863-9cc52105-7d83-4275-9c90-16a77b0e68de.png)

Another example from Console would be the Payment history page where they can download the receipts. 
![image](https://user-images.githubusercontent.com/72031986/130823481-d1143e69-c3ea-4b79-8a82-38cad7b77940.png)

Looking at the guidelines from [Polaris](https://polaris.shopify.com/content/actionable-language#section-export-vs-download) and [Carbon](https://www.carbondesignsystem.com/community/patterns/export-pattern/), they have similar ways to distinguish between Export and Download. Export implies that I'm transforming the data I'm currently viewing into a format that I need, whether thats a downloadable format or to another system. Download implies I'm downloading a document that is related to the data that I'm currently viewing such as a shipping label or a receipt or an invoice. 

Using terminology such as 'Export invoice' and 'Export payment receipt' in conformance with the pattern we have currently seems like we're going against the more widespread understanding of the these terms. 

cc @ctecks @andoyle

**Comment by TheSisb** [comment]
Hi @PoojMenon,

It looks like the data export pattern isn't very clear, thanks for bringing this to our attention. Is it this line that is causing the unclarity for you?

> The delayed export is an exception to this copy guideline and uses the word “Download” and the DownloadIcon instead.
https://paste.twilio.design/patterns/data-export#general 


Later in the document it is clarified that `Download` is in fact used the way you are suggesting:

> Send an email notification to the customer when the export is ready, with a link to or attachment of the file(s). In this second step, use the word “Download” (example: “Download CSV”) to trigger the actual file download since the customer already clicked an export action to start the export process.
https://paste.twilio.design/patterns/data-export#delayed-export

If you can help clarify which parts of the documents are unclear or misleading, it would help us a great deal in improving it. Thanks again!

**Comment by PoojMenon** [comment]
Hi @TheSisb 

My understanding after reading the pattern was that 'Export' is to be used for all immediate exports/downloads. And 'Download' for the 2nd step during a delayed export scenario. 

I also started this discussion as a follow up to UX review feedback (cc @katieporter) 

> Review the Data Export pattern to try to simplify the Export and Download actions throughout the experience. Use that as the source of truth around Export vs. Download language. In general, the pattern recommends that we always use the term "Export", unless you are downloading a file that has already been exported. This reduces cognitive load on the user so they don't have to anticipate the difference between exporting and downloading.

From what I understand from the above, the guidance is that I use 'Export invoice' instead of 'Download invoice' as per the pattern. But I don't think downloading an invoice is an export action.

I think in terms of the pattern itself, its possibly lacking the distinction between exporting a data set vs downloading related files.

**Reply by PoojMenon** [reply]
Bumping as I haven't heard back

**Comment by SiTaggart** [comment]
Hi @PoojMenon 

Without knowing the exact context of the UX Review feedback, as it might relate to specific screens or instances, my interpretation of the Export pattern is similar to yours.

When extracting data from Twilio systems into a new format or file, we are "exporting" the data.

When downloading an already existing file we are "downloading".

Invoices and receipts are generated documents from the business. They already exist as they are sent to the customer. You are not slicing or refining data to create a new export, but downloading something that already exists.

So to reiterate your initial post, in the first screen shot of a list view of invoices I would be exporting the list of invoices to a CSV, but downloading existing invoices. I think that's spot on and if I'm wrong, then we clearly have a problem with the documentation. Tagging in @twilio-labs/design-systems-pd for clarification.

I will say that the export pattern is potentially a little too focused on interacting with lists of data, and leaves common download situations to small references in a larger document. We might try to improve this to make it a little clearer in the future.

**Comment by kate-porter** [comment, marked as answer]
Hey @PoojMenon - thanks for bumping this one.

On another review of both the design and the Data Export pattern, I agree with you and Simon. I think we are lacking clarification in the documentation around downloading _related_ documents that aren't shown on the page. "Download" feels right for the invoices.

I'm going to chat with @twilio-labs/design-systems-pd about the best way to incorporate this feedback back into the Data Export pattern. This is our first time getting feedback on an existing pattern, so we don't quite have that workflow nailed down yet!

If you'd be at all interested in contributing this particular use case, please do let us know—that would be a huge help! And I imagine it wouldn't need to be a big lift—I think drafting up a paragraph on this use case would cover it, and then we could get it added as a variant on the pattern.

Let me know what ya think!

**Reply by PoojMenon** [reply]
Thanks Katie for the clarification. I'll keep 'download' for the invoices and I'm happy to contribute to the pattern. Feel free to DM me and let me know how to go about it.

**Reply by kate-porter** [reply]
Will do, thanks @PoojMenon!

**Comment by jweintraub** [comment]
The pattern that Sang Yeo worked on was in part based on this research...see lines 30, 33, & 36
https://docs.google.com/spreadsheets/d/1fcBBLYfouXPZp_o45mcYy6m5iKuZu36qTFSgxIcv2TA/edit#gid=1449337720

We prioritized "export" as the default wording in order to avoid inconsistency based on the user feedback as well as the competitive design systems research which we felt was not adequately describing our use cases. In  all our cases today, the data does need to be pulled (exported) and downloaded. However, if there are new cases outside of this, I'll leave it to you and the design systems team to determine that.

**Reply by PoojMenon** [reply]
This is really interesting. I do think this use case was specific to data on the page such as logs being exported into a format they wanted, for which Export definitely makes sense. 

I do think downloading related documents should probably be considered as a separate use case.
